### PR TITLE
bug/ISSUE149-border-not-rendered

### PR DESCRIPTION
**Pull request by janschaible**

### CODE REVIEW

**Comment by janschaible** [comment]
@FelixSchladt @B1TC0R3 should only the focused window have a border ?:)
If so, should the border be inwards or outwards ?:)

**Comment by FelixSchladt** [comment]
I think every window should have a border.
We have two different colors defined in our config.
The focused window will have a different color than the not focused one.

To have a hex color as bg color for the frame we will need to draw this with cairo like in the status bar.

**Comment by FelixSchladt** [comment]
This could be part of a follow up ticket

File: src/screeninfo/mod.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl ScreenInfo {
         }
 
         self.create_status_bar_window(event);
-        self.status_bar.as_mut().unwrap().draw();
+        self.status_bar.as_mut().unwrap().draw_without_border();
```
Comment (gungula) [inline src/screeninfo/mod.rs:156]: makes sense but i dont really get why the statusbar didnt have any borders around it

File: src/windowstate.rs [hunk 2]
```diff
@@ -124,15 +129,51 @@ impl WindowState {
...
+            .unwrap();
+
+        self.connection.map_window(self.frame).unwrap();
+        self.connection.map_window(self.window).unwrap();
+
+        self.connection.ungrab_server().unwrap();
+        self.connection.flush().unwrap();
+    }
+
+    pub fn draw_without_border(&self){
```
Comment (gungula) [inline src/windowstate.rs:159]: why do we have three functions?

I would think we only need two. One normal ones with gaps and frames, for fullscreen and status bar.
Another one with gaps and frames for normal windows.

File: src/windowstate.rs [hunk 3]
```diff
@@ -124,15 +129,51 @@ impl WindowState {
...
 
-        let window_aux = ConfigureWindowAux::new()
+        let frame_aux = ConfigureWindowAux::new()
             .x(self.x)
             .y(self.y)
             .width(self.width)
             .height(self.height);
 
+        let window_aux = ConfigureWindowAux::new()
+            .x(0)
```
Comment (gungula) [inline src/windowstate.rs:140]: why is this hardcoded?

File: src/windowstate.rs [hunk 4]
```diff
@@ -156,8 +195,8 @@ impl WindowState {
             .height(self.height - (self.gap_size * 2));
 
         let window_aux = ConfigureWindowAux::new()
-            .x(self.x + (self.border_width + self.gap_size) as i32)
-            .y(self.y + (self.border_width + self.gap_size) as i32)
+            .x(self.border_width as i32)
```
Comment (gungula) [inline src/windowstate.rs:198]: what happened to the gaps?

File: src/workspace/mod.rs [hunk 5]
```diff
@@ -488,3 +508,11 @@ impl Workspace {
         }
     }
 }
+
+fn draw_window(window:&WindowState, focused_window: Option<u32>){
+    if Some(window.window) == focused_window {
+        window.draw();
+    }else{
+        window.draw_without_border();
```
Comment (gungula) [inline src/workspace/mod.rs:516]: all windows should have a border, focused or not.

The color of the border should differentiate.

I would postpone this though for now and implement this later.
The focusing could be given with a parameter. WDY think?

For now we could do black and white pixel for unfocused/focused?

**Review by gungula — CHANGES_REQUESTED** [review]
teamwork with @FelixSchladt

**Comment by FelixSchladt** [comment]
there is a poc using the border width attribute. 
5f669620c77b2a24c638c7fd196ced990d2fae65

This should fix the focus issues of the current solution and easy handling by not requiring a seperate frame window to be created...

**Comment by FelixSchladt** [comment]
closing this pr as code is more a workaround and does not provide the desired functionality

**Pull request closed without merging**